cmd/kubeadm/app/phases/selfhosting/selfhosting_test.go

```diff
@@ -0,0 +1,544 @@
+        - --proxy-client-cert-file=/etc/kubernetes/pki/front-proxy-client.crt
+        - --experimental-bootstrap-token-auth=true
+        - --requestheader-group-headers=X-Remote-Group
+        - --requestheader-allowed-names=front-proxy-client
+        - --kubelet-client-key=/etc/kubernetes/pki/apiserver-kubelet-client.key
+        - --requestheader-client-ca-file=/etc/kubernetes/pki/front-proxy-ca.crt
+        - --proxy-client-key-file=/etc/kubernetes/pki/front-proxy-client.key
+        - --kubelet-preferred-address-types=InternalIP,ExternalIP,Hostname
+        - --requestheader-username-headers=X-Remote-User
+        - --authorization-mode=Node,RBAC
+        - --advertise-address=192.168.200.101
+        - --etcd-servers=http://127.0.0.1:2379
+        image: gcr.io/google_containers/kube-apiserver-amd64:v1.7.0
+        livenessProbe:
+          failureThreshold: 8
+          httpGet:
+            host: 127.0.0.1
+            path: /healthz
+            port: 6443
+            scheme: HTTPS
+          initialDelaySeconds: 15
+          timeoutSeconds: 15
+        name: kube-apiserver
+        resources:
+          requests:
+            cpu: 250m
+        volumeMounts:
+        - mountPath: /etc/kubernetes
+          name: k8s
+          readOnly: true
+        - mountPath: /etc/ssl/certs
+          name: certs
+        - mountPath: /etc/pki
+          name: pki
+      dnsPolicy: ClusterFirstWithHostNet
+      hostNetwork: true
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      tolerations:
+      - effect: NoSchedule
+        key: node-role.kubernetes.io/master
+      volumes:
+      - hostPath:
+          path: /etc/kubernetes
+        name: k8s
+      - hostPath:
+          path: /etc/ssl/certs
+        name: certs
+      - hostPath:
+          path: /etc/pki
+        name: pki
+  updateStrategy: {}
+status:
+  currentNumberScheduled: 0
+  desiredNumberScheduled: 0
+  numberMisscheduled: 0
+  numberReady: 0
+`
+
+	testControllerManagerPod = `
+apiVersion: v1
+kind: Pod
+metadata:
+  annotations:
+    scheduler.alpha.kubernetes.io/critical-pod: ""
+  creationTimestamp: null
+  labels:
+    component: kube-controller-manager
+    tier: control-plane
+  name: kube-controller-manager
+  namespace: kube-system
+spec:
+  containers:
+  - command:
+    - kube-controller-manager
+    - --service-account-private-key-file=/etc/kubernetes/pki/sa.key
+    - --cluster-signing-cert-file=/etc/kubernetes/pki/ca.crt
+    - --cluster-signing-key-file=/etc/kubernetes/pki/ca.key
+    - --leader-elect=true
+    - --kubeconfig=/etc/kubernetes/controller-manager.conf
+    - --controllers=*,bootstrapsigner,tokencleaner
+    - --root-ca-file=/etc/kubernetes/pki/ca.crt
+    - --address=127.0.0.1
+    - --use-service-account-credentials=true
+    image: gcr.io/google_containers/kube-controller-manager-amd64:v1.7.0
+    livenessProbe:
+      failureThreshold: 8
+      httpGet:
+        host: 127.0.0.1
```

Comment:
Yeah, but my point is that the scheduler and controller manager don't need to be on host network. For apiserver (which is on host network), leaving blank will still work because it'll just ping that pod's API